config/reader.go

```diff
@@ -9,16 +9,16 @@ import (
 )
 
 var (
-	product *ProductConfig
+	product *Product
 	once    sync.Once
 )
 
-type ProductConfig struct {
+type Product struct {
```

Comment:
Not sure for others but to me these references could cause possible confusions down the line as we have a struct called `Product` and it contains string var called `Product`.. and in SU we discussed about adding a package called `product.go`... wondering if there is any way to incorporate `Product` related stuff together and also come up with better naming conventions

Reply:
lets review this PR only within the scope.
what should we naming the product (rke2,k3s) config then >?

Reply:
Sure we can discuss about naming conventions and possible resolutions in separate PR